pkg/attestation/crafter/materials/oci_image_test.go

```diff
@@ -0,0 +1,295 @@
+			require.NotNil(t, containerImage)
+			assert.Equal(t, tc.wantName, containerImage.Name)
+			assert.Equal(t, tc.wantTag, containerImage.Tag)
+			assert.NotEmpty(t, containerImage.Digest)
+			assert.True(t, len(containerImage.Digest) > 0, "digest should not be empty")
+		})
+	}
+}
+
+func TestOCIImageCraft_LayoutWithDigestSelector(t *testing.T) {
+	testCases := []struct {
+		name        string
+		setupLayout func(t *testing.T) (string, string) // returns (layoutPath, digestToSelect)
+		wantErr     string
+		wantName    string
+		wantTag     string
+	}{
+		{
+			name: "select second image by digest",
+			setupLayout: func(t *testing.T) (string, string) {
+				layoutPath, digests := createTestOCILayoutMultiple(t, []imageSpec{
+					{name: "first-image", tag: "v1.0.0"},
+					{name: "second-image", tag: "v2.0.0"},
+				})
+				return layoutPath, digests[1] // Select second image
+			},
+			wantName: "second-image",
+			wantTag:  "v2.0.0",
+		},
+		{
+			name: "digest not found",
+			setupLayout: func(t *testing.T) (string, string) {
+				layoutPath, _ := createTestOCILayoutMultiple(t, []imageSpec{
+					{name: "test-image", tag: "v1.0.0"},
+				})
+				return layoutPath, "sha256:nonexistent"
+			},
+			wantErr: "not found in OCI layout",
+		},
+		{
+			name: "multiple images without digest selector",
+			setupLayout: func(t *testing.T) (string, string) {
+				layoutPath, _ := createTestOCILayoutMultiple(t, []imageSpec{
+					{name: "first-image", tag: "v1.0.0"},
+					{name: "second-image", tag: "v2.0.0"},
+					{name: "third-image", tag: "v3.0.0"},
+				})
+				return layoutPath, "" // No digest selector
+			},
+			wantErr: "contains 3 images, please specify which one",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			layoutPath, digest := tc.setupLayout(t)
+			imageRef := layoutPath
+			if digest != "" {
+				imageRef = layoutPath + "@" + digest
+			}
+
+			schema := &contractAPI.CraftingSchema_Material{
+				Name: "test",
+				Type: contractAPI.CraftingSchema_Material_CONTAINER_IMAGE,
+			}
+			l := zerolog.Nop()
+			crafter, err := materials.NewOCIImageCrafter(schema, nil, &l)
+			require.NoError(t, err)
+
+			got, err := crafter.Craft(context.TODO(), imageRef)
+			if tc.wantErr != "" {
+				assert.ErrorContains(t, err, tc.wantErr)
+				return
+			}
+
+			require.NoError(t, err)
+			assert.Equal(t, contractAPI.CraftingSchema_Material_CONTAINER_IMAGE.String(), got.MaterialType.String())
+
+			// Check container image fields
+			containerImage := got.GetContainerImage()
+			require.NotNil(t, containerImage)
+			assert.Equal(t, tc.wantName, containerImage.Name)
+			assert.Equal(t, tc.wantTag, containerImage.Tag)
+			assert.NotEmpty(t, containerImage.Digest)
+		})
+	}
+}
+
+type imageSpec struct {
+	name string
+	tag  string
+}
+
+// createTestOCILayoutMultiple creates an OCI layout with multiple images for testing
+func createTestOCILayoutMultiple(t *testing.T, specs []imageSpec) (string, []string) {
```

Comment:
alternatively you could add image layouts to `testdata` and read it from there, that's what I believe we do for most other material types, have an offline, actual piece of evidence. 

If you go that route, you could even add different copies from different tools, if you see value.